packages/suite/src/views/wallet/trading/common/TradingSelectedOffer/TradingReceiveAddress/TradingReceiveAddressModal.tsx

```diff
@@ -0,0 +1,171 @@
+import { spacings } from '@trezor/theme';
+
+import { Translation } from 'src/components/suite/Translation';
+import { useTranslation } from 'src/hooks/suite';
+import { TradingVerifyFormProps } from 'src/types/trading/tradingVerify';
+import { TradingExtraField } from 'src/views/wallet/trading/common/TradingSelectedOffer/TradingReceiveAddress/TradingExtraField';
+import { useTradingReceiveAddressValues } from 'src/views/wallet/trading/common/TradingSelectedOffer/TradingReceiveAddress/TradingReceiveAddress';
+import { useReceiveAddressModalControls } from 'src/views/wallet/trading/common/TradingSelectedOffer/TradingReceiveAddress/useReceiveAddressModalControls';
+
+export const TradingReceiveAddressModal = () => {
+    const { tradingReceiveAddress, cryptoId, extraFieldDescription } =
+        useTradingReceiveAddressValues();
+    const modalControls = useReceiveAddressModalControls();
+
+    const { translationString } = useTranslation();
+    const { cryptoIdToPlatformName, cryptoIdToCoinName, cryptoIdToNativeCoinSymbol } =
+        useTradingInfo();
+
+    const { networkId, contractAddress } = parseCryptoId(cryptoId);
+    const networkName = contractAddress
+        ? cryptoIdToPlatformName(networkId)
+        : cryptoIdToCoinName(networkId);
+
+    const { selectedAccountOption, selectAccountOptions } = tradingReceiveAddress;
+
+    const form = useForm<TradingVerifyFormProps>({
+        mode: 'onChange',
+    });
+
+    const requiresExtraField = !!extraFieldDescription;
+
+    const { ref: networkRef, ...networkField } = form.register('address', {
+        required: translationString('TR_EXCHANGE_RECEIVING_ADDRESS_REQUIRED'),
+        validate: value => {
+            if (cryptoId) {
+                const symbol =
+                    cryptoIdToNetwork(cryptoId)?.symbol ?? cryptoIdToNativeCoinSymbol(cryptoId);
+                if (value && !addressValidator.validate(value, symbol)) {
+                    return translationString('TR_EXCHANGE_RECEIVING_ADDRESS_INVALID');
+                }
+            }
+        },
+    });
+
+    const { ref: descriptionRef, ...descriptionField } = form.register('extraField', {
+        required: extraFieldDescription?.required
+            ? translationString('TR_EXCHANGE_EXTRA_FIELD_REQUIRED', {
+                  extraFieldName: extraFieldDescription.name,
+              })
+            : undefined,
+        validate: value => {
+            let valid = true;
+            if (value) {
+                if (extraFieldDescription?.type === 'hex') {
+                    valid = isHexValid(value);
+                } else if (extraFieldDescription?.type === 'number') {
+                    valid = isInteger(value);
+                }
+            }
+            if (!valid) {
+                return translationString('TR_EXCHANGE_EXTRA_FIELD_INVALID', {
+                    extraFieldName: extraFieldDescription?.name,
+                });
+            }
```

Comment:
```suggestion
            const { type, name: extraFieldName } = extraFieldDescription ?? {};
            const validators = {
                hex: isHexValid,
                number: isInteger,
                text: () => true,
            };

            const valid = type ? validators[type](value) : true;

            if (!valid) {
                return translationString('TR_EXCHANGE_EXTRA_FIELD_INVALID', {
                    extraFieldName,
                });
            }

```

Just something for consideration, no need to update it. I just like to omit using `let` variables when not necessary.